### PR TITLE
User can now get directions to a building. Also including unit tests for building popup.

### PR DESCRIPTION
**Pull request by felixlapierre**
There is a slight issue with this solution:

The google map's popup is rendered as an image instead of as an Android view. Because of this, it's not possible to attach click listeners to the button - you can only attach click listeners to the entire window. As such, clicking anywhere in the window will start the directions flow.

Mike did a lot of investigation about the workaround and we agreed that this is the best we can do without creating a lot of work. If it's deemed necessary we can fix the problem later.

### CODE REVIEW

File: app/src/main/java/com/example/campusguide/directions/DirectionsFlow.kt [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+            val destinationLatLng = LatLng(destination.latitude, destination.longitude)
+            cont.resume(destinationLatLng.toString())
+        }
+        chooseDestinationOptions.show(
+            activity.supportFragmentManager,
+            "chooseDestinationOptions"
+        )
+    }
+
+    private suspend fun getOrigin() = suspendCoroutine<String> { cont ->
```
Comment (bogalla) [inline app/src/main/java/com/example/campusguide/directions/DirectionsFlow.kt]: what do these 2 functions do?

Comment (bogalla) [reply app/src/main/java/com/example/campusguide/directions/DirectionsFlow.kt]: getOrigin and getDestination

**Comment by bogalla** [comment]
whats with the weird linux file?

**Comment by bogalla** [comment]
fix merge conflicts

File: app/src/main/java/com/example/campusguide/directions/DirectionsFlow.kt (hunk 1 above)
Comment (felixlapierre) [reply app/src/main/java/com/example/campusguide/directions/DirectionsFlow.kt]: Each creates their popup that asks the user to enter origin / destination (current location, calendar event, search for location, etc). Thanks to the power of coroutines, the function will return once the popup is closed :)

**Review by pobudzey — CHANGES_REQUESTED** [review]
Everything works as expected and all unit tests are passing
Fix conflicts and we're good to go

**Review by pobudzey — APPROVED** [review]

**Pull request merged**